### PR TITLE
Fix manual failover/switchover checks

### PR DESCRIPTION
**Pull request by hughcapet**
In case of manual failover/switchover failover possibility should be checked only against the candidate

### CODE REVIEW

File: patroni/ha.py [hunk 1]
```diff
@@ -1189,12 +1193,13 @@ def process_manual_failover_from_leader(self) -> Optional[str]:
                     logger.warning('Failover is possible only to a specific candidate in a paused state')
                 else:
                     if self.is_synchronous_mode():
-                        if failover.candidate and not self.cluster.sync.matches(failover.candidate):
+                        # every sync_standby or the cnadidate if is in sync_standbys
```
Comment (CyberDem0n) [inline patroni/ha.py:1196]:
```suggestion
                        # every sync_standby or the candidate if is in sync_standbys
```

File: patroni/ha.py [hunk 2]
```diff
@@ -1085,7 +1085,11 @@ def on_shutdown(checkpoint_location: int) -> None:
             # It could happen if Postgres is still archiving the backlog of WAL files.
             # If we know that there are replicas that received the shutdown checkpoint
             # location, we can remove the leader key and allow them to start leader race.
-            if self.is_failover_possible(self.cluster.members, cluster_lsn=checkpoint_location):
+            failover = self.cluster.failover
+            # for a manual failover/switchover with a candidate, we should check the requested candidate only
+            members = [m for m in self.cluster.members if not failover or not failover.candidate
+                       or failover.candidate and m.name == failover.candidate]
```
Comment (CyberDem0n) [inline patroni/ha.py:1091]: I think this check is redundant.
```suggestion
                       or m.name == failover.candidate]
```

**Comment by CyberDem0n** [comment]
:+1:

**Comment by hughcapet** [comment]
👍

**Pull request merged**